drop-boxes/register-nanopore-dropbox/register-nanopore.py

```diff
@@ -132,44 +129,75 @@ def createExperimentFromMeasurement(transaction, currentPath, space, project, me
     runExperiment.setPropertyValue("Q_NANOPORE_HOSTNAME", measurement.getMachineHost())
     runExperiment.setPropertyValue("Q_DATA_GENERATION_FACILITY", origin)
     runExperiment.setPropertyValue("Q_MEASUREMENT_START_DATE", convertTime(measurement.getStartDate()))
-    #if measurement.getAdapter():
-    #    runExperiment.setPropertyValue("Q_SEQUENCING_ADAPTER", measurement.getAdapter())
-    # runExperiment.setPropertyValue("Q_EXTERNALDB_ID",) best skip and parse sample information at sample level, no experiment-wide ID from what I can tell
+    if measurement.getAdapter():
+        runExperiment.setPropertyValue("Q_SEQUENCING_ADAPTER", measurement.getAdapter())
     # handle measured samples
     for barcode in rawDataPerSample.keySet():
+        print "handling barcode: "+barcode
         datamap = rawDataPerSample.get(barcode)
         newLogFolder = createLogFolder(currentPath)
         copyLogFilesTo(measurement.getLogFiles(), currentPath, newLogFolder)
         createSampleWithData(transaction, space, barcode, datamap, runExperiment, currentPath, newLogFolder)
 
+# fills the global dictionary containing all checksums for paths from the global checksum file
+# 20886d8be5b6e899c707292f942ccb2edb3dd5dd69df757ed029d9f3200fa95a *20200304185641_QNANO038AT_E19D023c02_PAE34304/20200219_1107_2-A3-D3_PAE34304_6351def9/fastq_fail/PAE34...
+def fillChecksumMap(checksumFilePath):
+    with open(checksumFilePath, 'r') as chf:
+        for line in chf:
+            tokens = line.strip().split(" *")
+            print tokens
+            checksumMap[tokens[1]] = tokens[0]
+
+# creates a file containing checksums and paths for files contained in the passed path using the global checksum dictionary
+def createChecksumFileForFolder(incomingPath, folderPath):
+
+    relativePath = os.path.relpath(folderPath, incomingPath)
+
+    pathEnd = os.path.basename(os.path.normpath(folderPath))
+    checksumFile = os.path.join(folderPath, pathEnd+'.sha256sum')
+
+    with open(checksumFile, 'w') as f:
+        for key, value in checksumMap.items():
+            if key.startswith(relativePath):
+                f.write(value+' *'+key+'\n')
+
+def prepareDataFolder(incomingPath, currentPath, targetPath, dataObject, suffix):
+    name = dataObject.getName()
+    rel = dataObject.getRelativePath()
+    src = os.path.join(os.path.dirname(currentPath), rel)
```

Comment:
what is `src`? I presume it is some kind of source but of what and how does it belong here?

Reply:
added comments

Reply:
I guess this is changed now?